File: leetcode_solutions/0274-h-index/solution.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        citations.sort(reverse=True)
        h = 0
        for i, cite in enumerate(citations):
                if cite >= i + 1:
                    h+=1
        return h
        

File: leetcode_solutions/0274-h-index/solution.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        sort(citations.begin(), citations.end(), greater<>());
        int n = citations.size();
        int h = 0;
        for (int i = 0; i < citations.size(); i++){
            if (citations[i] >= i + 1){
                // citations of paper >= number of papers cited before it x amount of times
                h++;
            }
        }

        return h;
    }
};
